src/utils/availability_meeting_types.ts

```diff
@@ -0,0 +1,135 @@
+import { MeetingType } from '@/types/Account'
+
+import { initDB } from './database'
+import { AvailabilityBlockNotFoundError } from './errors'
+
+const db = initDB()
+
+export const getMeetingTypesForAvailabilityBlock = async (
+  account_address: string,
+  availability_block_id: string
+): Promise<MeetingType[]> => {
+  // First verify the availability block exists and belongs to the account
+  const { data: block, error: blockError } = await db.supabase
+    .from('availabilities')
+    .select('id')
+    .eq('id', availability_block_id)
+    .eq('account_owner_address', account_address)
+    .single()
+
+  if (blockError || !block) {
+    throw new AvailabilityBlockNotFoundError()
+  }
+
+  // Get meeting type IDs associated with this availability block
+  const { data: meetingTypeIds, error: idsError } = await db.supabase
+    .from('meeting_type_availabilities')
+    .select('meeting_type_id')
+    .eq('availability_id', availability_block_id)
```

Comment:
You could simply use a join to meeting_type here so we don't need a duplicate request